lib/search.js

```diff
@@ -66,35 +57,59 @@ function search (args, silent, staleness, cb) {
     return s.toLowerCase()
   })
 
-  getFilteredData(staleness, opts, searchexclude, function (er, data) {
-    // now data is the list of data that we want to show.
-    // prettify and print it, and then provide the raw
-    // data to the cb.
-    if (er || silent) return cb(er, data)
-    output(prettify(data, args))
-    cb(null, data)
+  updateIndex(staleness, function (er, stream) {
```

Comment:
I think this callback should be named and hoisted, given its length, complexity, and the fact there are further callbacks nested in it.